### DISCUSSION TITLE
Conditional Validation

### DISCUSSION THREAD

**Discussion by derekzernach** [dynamoose/dynamoose#1157] [Q&A]
Is there a way to only run validation on an attribute if a different attribute is set to a specific value?

In a different package that i'm using, it uses Joi where I can do this...

`age: Joi.when('application_id', {
      is: 'foo',
      then: Joi.string().trim().required(),
    }),`

So this attribute is only required if `application_id` is set to `foo`

**Comment by fishcharlie** [comment]
@derekzernach Not currently. Sounds like an **amazing** idea tho.

I wonder if we just want to add a feature where you can pass in a `Joi.object` schema instance to your Dynamoose schema? Then the last step in validation will be checking that Joi validation?

My only thing is I don't really wanna have Joi as a dependency. So it'd require some work around having the Joi require statement be optional and have it be dependent on if the Dynamoose user has Joi as a dependency.

Feel free to create an issue for this.

**Reply by derekzernach** [reply]
Awesome! The very out dated package I'm using is `https://github.com/clarkie/dynogels` that I went to years ago because it offered some things dynamoose didn't. Looking to make the switch back to dynamoose if we can get validation like this working whether it uses Joi or something else.
